frontend/src/state/reducers/scheduleReducer.ts

```diff
@@ -134,6 +142,17 @@ export const scheduleReducer = (
         draft.present = JSON.parse(JSON.stringify(draft.past));
         return draft;
       }
+      case getType(removeTransferClassAction): {
+        const { course } = action.payload;
+        // save prev state with a deep copy
+        draft.past = JSON.parse(JSON.stringify(draft.present));
```

Comment:
We may want to look into a different method of deep copying here, I'm not sure if it is safe to use JSON stringify/parse on a `ScheduleStateSlice` without potentially losing data since it contains `closedYears` which is a `Set`.

https://stackoverflow.com/questions/122102/what-is-the-most-efficient-way-to-deep-clone-an-object-in-javascript

Reply:
Good catch!! I'm going to keep stringify for now since I'm having issue with cloneDeep. Iman's going to make a ticket for it in the backlog :)